test/unittests/test_dd_synthesis.cpp

```diff
@@ -0,0 +1,72 @@
+#include "algorithms/synthesis/dd_synthesis.hpp"
+#include "algorithms/synthesis/encoding.hpp"
+#include "core/io/pla_parser.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TestDDSynth: public testing::TestWithParam<std::string> {
+protected:
+    TruthTable                     tt{};
+    std::string                    testCircuitsDir = "./circuits/";
+    std::unique_ptr<dd::Package<>> ddSynth         = std::make_unique<dd::Package<>>(9U);
+    std::string                    fileName;
+
+    void SetUp() override {
+        fileName = testCircuitsDir + GetParam() + ".pla";
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(TestDDSynth, TestDDSynth,
+                         testing::Values(
+                                 "cnot",
+                                 "swap",
+                                 "toffoli",
+                                 "x2Bit",
+                                 "test_dd_synthesis_1",
+                                 "test_dd_synthesis_2",
+                                 "3_17_6",
+                                 "bitwiseXor2Bit",
+                                 "adder2Bit",
+                                 "adder3Bit",
+                                 "4_49_7",
+                                 "hwb4_12",
+                                 "hwb5_13",
+                                 "hwb6_14",
+                                 "hwb7_15",
+                                 "hwb8_64",
+                                 "hwb9_65"),
```

Comment:
Do we really need that many instances of the same benchmark?

Reply:
All these benchmarks are worst case scenario benchmarks. In the previous days, these helped me to debug and  improve the synthesis related functions. I would prefer to keep them. However, let me know your opinion on this.

Reply:
Alright, makes sense 👍🏻 feel free to resolve this conversation then